File: photo_app_backend/src/main/java/com/studiophoto/photoappbackend/model/ContactInfo.java
package com.studiophoto.photoappbackend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "contact_info")
public class ContactInfo {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String address;
    private String phoneNumber;
    private String whatsappNumber;
    private String email;
    private String openingHours;
    private String facebookUrl;
    private String twitterUrl;
    private String instagramUrl;
    private String linkedinUrl;
}

File: photo_app/lib/models/contact_info.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:flutter/foundation.dart';
part 'contact_info.freezed.dart';
part 'contact_info.g.dart';

@freezed
class ContactInfo with _$ContactInfo {
  const factory ContactInfo({
    required String address,
    required String phoneNumber,
    String? whatsappNumber,
    required String email,
    required String openingHours,
    String? facebookUrl,
    String? twitterUrl,
    String? instagramUrl,
    String? linkedinUrl,
  }) = _ContactInfo;

  factory ContactInfo.fromJson(Map<String, Object?> json) => _$ContactInfoFromJson(json);
}
